### PR TITLE
[FTL-2291] Usage of url-resolver package in SDK

### PR DESCRIPTION
**Pull request by vladmitchenko**

### CODE REVIEW

**Review by oleksiipiliugin — APPROVED** [review]

**Review by anton-iskryzhytskyi — DISMISSED** [review]
LGTM

**Review by inga-affinidi — DISMISSED** [review]

**Review by anton-iskryzhytskyi — APPROVED** [review]

File: sdk/core/src/services/WalletStorageService.ts [hunk 1]
```diff
@@ -78,7 +77,7 @@ export default class WalletStorageService {
     credentialOfferResponseToken: string,
     options: { keyStorageUrl?: string; issuerUrl?: string; accessApiKey?: string; apiKey?: string } = {},
   ): Promise<SignedCredential[]> {
-    const keyStorageUrl = options.keyStorageUrl || STAGING_KEY_STORAGE_URL
+    const keyStorageUrl = options.keyStorageUrl || resolveUrl(Service.KEY_STORAGE, 'staging')
```
[the remark this answers is not in the document]
Comment (vitalyrudenko-affinidi) [reply sdk/core/src/services/WalletStorageService.ts:80]: How do we provide production/dev key storage url then?

[the remark this answers is not in the document]
Comment (vladmitchenko) [reply sdk/core/src/services/WalletStorageService.ts:80]: keyStorageUrl provided from basic options which are already resolved with resolveUrl
Question is: "do we really need  `|| resolveUrl(Service.KEY_STORAGE, 'staging')`?

[the remark this answers is not in the document]
Comment (vitalyrudenko-affinidi) [reply sdk/core/src/services/WalletStorageService.ts:80]: I don't get it. If we hardcode 'staging', how can we provide anything but staging?

[the remark this answers is not in the document]
Comment (vladmitchenko) [reply sdk/core/src/services/WalletStorageService.ts:80]: As I understood from the code `STAGING_KEY_STORAGE_URL ` was used as reserved one (for some reasons I'm not sure what reasons and don't want to break some logic with removing this part)
If check the code `keyStorageUrl` value is provided by `getOptionsFromEnvironment` function, and there we use `resolveUrl` function with using not hard coded `env`

[the remark this answers is not in the document]
Comment (cosminm-affnd) [reply sdk/core/src/services/WalletStorageService.ts:80]: Thinking about it do we even need the second parameter?  `resolveUrl(Service.KEY_STORAGE)` should suffice, everything else can be implemented in the Resolver class.

[the remark this answers is not in the document]
Comment (vladmitchenko) [reply sdk/core/src/services/WalletStorageService.ts]: Removed this reserved staging url, cuz it was only for test, and in test file just provided url as param

[the remark this answers is not in the document]
Comment (vitalyrudenko-affinidi) [reply sdk/core/src/services/WalletStorageService.ts]: @cosminm-affnd I'm a bit confused. How would it know which environment we want to use if we don't provide it?

[the remark this answers is not in the document]
Comment (inga-affinidi) [reply sdk/core/src/services/WalletStorageService.ts]: Ideally we would make `keyStorageUrl` (along with other parameters) non-optional here.

It is an internal class, it should not be used by SDK consumers; it is initialized in `CommonNetworkMember`, and we know there exactly what `keyStorageUrl` we want to use.

Default staging url was only used here as a fallback, probably for tests. It is the fallback that should be removed, not the actual configuration.

Note that it's not only environment that can be different; SDK consumers can supply an URL of their own key storage service.

**Review by anton-iskryzhytskyi — APPROVED** [review]

**Review by YuriiSkrypnyk — APPROVED** [review]
LGTM

**Review by anton-iskryzhytskyi — APPROVED** [review]

**Pull request merged**